### ISSUE TITLE
Bug: VPN failing healthcheck with custom Wireguard

### ISSUE THREAD

**Issue by diamondsw** [qdm12/gluetun#1955]
### Is this urgent?

No

### Host OS

Debian 11.8

### CPU arch

x86_64

### VPN service provider

Custom

### What are you using to run the container

docker-compose

### What is the version of Gluetun

Running version latest built on 2023-11-08T10:11:52.918Z (commit 8318be3)

### What's the problem 🤔

Cannot connect to custom selfhosted Wireguard VPN (created via Linuxserver.io image). Identical wg0.conf connects without issue using the Linuxserver.io image. This certainly could be user error - I'm relatively new to gluetun and wireguard, although I have painful experience with IPSEC - but since I'm using an identical working wg0.conf, I'm having a hard time seeing where I may have gone wrong.

Any help is appreciated.

`docker-compose.yml` provided below.
`docker-compose up` output provided below.

Healthchecks in log repeat similarly, increasing timeout by 5s each check, no change. I've shelled into the container and likewise there's no working connectivity (ping, wget, etc).

Same wg0.conf used with `linuxserver/wireguard:latest`:
`docker compose up` output:
```
[migrations] started
[migrations] no migrations found
───────────────────────────────────────

      ██╗     ███████╗██╗ ██████╗ 
      ██║     ██╔════╝██║██╔═══██╗
      ██║     ███████╗██║██║   ██║
      ██║     ╚════██║██║██║   ██║
      ███████╗███████║██║╚██████╔╝
      ╚══════╝╚══════╝╚═╝ ╚═════╝ 

   Brought to you by linuxserver.io
───────────────────────────────────────

To support the app dev(s) visit:
WireGuard: https://www.wireguard.com/donations/

To support LSIO projects visit:
https://www.linuxserver.io/donate/

───────────────────────────────────────
GID/UID
───────────────────────────────────────

User UID:    1000
User GID:    1000
───────────────────────────────────────

Uname info: Linux 0523d9331798 5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux
**** It seems the wireguard module is already active. Skipping kernel header install and module compilation. ****
**** As the wireguard module is already active you can remove the SYS_MODULE capability from your container run/compose. ****
**** Client mode selected. ****
[custom-init] No custom files found, skipping...
**** Disabling CoreDNS ****
Warning: `/config/wg0.conf' is world accessible
[#] ip link add wg0 type wireguard
[#] wg setconf wg0 /dev/fd/63
[#] ip -4 address add 10.13.13.2 dev wg0
[#] ip link set mtu 1420 up dev wg0
[#] resolvconf -a wg0 -m 0 -x
s6-rc: fatal: unable to take locks: Resource busy
[#] wg set wg0 fwmark 51820
[#] ip -4 route add 0.0.0.0/0 dev wg0 table 51820
[#] ip -4 rule add not fwmark 51820 table 51820
[#] ip -4 rule add table main suppress_prefixlength 0
[#] iptables-restore -n
[ls.io-init] done.
```

Shelling into the container indicates it's working; checking public IP returns the VPN server endpoint as expected.

### Share your logs (at least 10 lines)

```plain text
[+] Running 1/0
 ⠿ Container gluetun  Created                                                                                                                                                                      0.0s
Attaching to gluetun
gluetun  | ========================================
gluetun  | ========================================
gluetun  | =============== gluetun ================
gluetun  | ========================================
gluetun  | =========== Made with ❤️ by ============
gluetun  | ======= https://github.com/qdm12 =======
gluetun  | ========================================
gluetun  | ========================================
gluetun  | 
gluetun  | Running version latest built on 2023-11-08T10:11:52.918Z (commit 8318be3)
gluetun  | 
gluetun  | 🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
gluetun  | 🐛 Bug? https://github.com/qdm12/gluetun/issues/new
gluetun  | ✨ New feature? https://github.com/qdm12/gluetun/issues/new
gluetun  | ☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
gluetun  | 💻 Email? quentin.mcgaw@gmail.com
gluetun  | 💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
gluetun  | 2023-11-10T16:05:24Z INFO [routing] default route found: interface eth0, gateway 172.30.128.1, assigned IP 172.30.128.2 and family v4
gluetun  | 2023-11-10T16:05:24Z INFO [routing] local ethernet link found: eth0
gluetun  | 2023-11-10T16:05:24Z INFO [routing] local ipnet found: 172.30.128.0/24
gluetun  | 2023-11-10T16:05:24Z INFO [firewall] enabling...
gluetun  | 2023-11-10T16:05:25Z INFO [firewall] enabled successfully
gluetun  | 2023-11-10T16:05:26Z INFO [storage] merging by most recent 17689 hardcoded servers and 17689 servers read from /gluetun/servers.json
gluetun  | 2023-11-10T16:05:27Z INFO Alpine version: 3.18.4
gluetun  | 2023-11-10T16:05:27Z INFO OpenVPN 2.5 version: 2.5.8
gluetun  | 2023-11-10T16:05:27Z INFO OpenVPN 2.6 version: 2.6.5
gluetun  | 2023-11-10T16:05:27Z INFO Unbound version: 1.17.1
gluetun  | 2023-11-10T16:05:27Z INFO IPtables version: v1.8.9
gluetun  | 2023-11-10T16:05:27Z INFO Settings summary:
gluetun  | ├── VPN settings:
gluetun  | |   ├── VPN provider settings:
gluetun  | |   |   ├── Name: custom
gluetun  | |   |   └── Server selection settings:
gluetun  | |   |       ├── VPN type: wireguard
gluetun  | |   |       └── Wireguard selection settings:
gluetun  | |   |           ├── Endpoint IP address: [REDACTED]
gluetun  | |   |           ├── Endpoint port: [REDACTED]
gluetun  | |   |           └── Server public key: [REDACTED]
gluetun  | |   └── Wireguard settings:
gluetun  | |       ├── Private key: [REDACTED]
gluetun  | |       ├── Interface addresses:
gluetun  | |       |   └── 10.13.13.2/32
gluetun  | |       ├── Allowed IPs:
gluetun  | |       |   ├── 0.0.0.0/0
gluetun  | |       |   └── ::/0
gluetun  | |       └── Network interface: tun0
gluetun  | |           └── MTU: 1400
gluetun  | ├── DNS settings:
gluetun  | |   ├── Keep existing nameserver(s): no
gluetun  | |   ├── DNS server address to use: 127.0.0.1
gluetun  | |   └── DNS over TLS settings:
gluetun  | |       ├── Enabled: yes
gluetun  | |       ├── Update period: every 24h0m0s
gluetun  | |       ├── Unbound settings:
gluetun  | |       |   ├── Authoritative servers:
gluetun  | |       |   |   └── cloudflare
gluetun  | |       |   ├── Caching: yes
gluetun  | |       |   ├── IPv6: no
gluetun  | |       |   ├── Verbosity level: 1
gluetun  | |       |   ├── Verbosity details level: 0
gluetun  | |       |   ├── Validation log level: 0
gluetun  | |       |   ├── System user: root
gluetun  | |       |   └── Allowed networks:
gluetun  | |       |       ├── 0.0.0.0/0
gluetun  | |       |       └── ::/0
gluetun  | |       └── DNS filtering settings:
gluetun  | |           ├── Block malicious: yes
gluetun  | |           ├── Block ads: no
gluetun  | |           ├── Block surveillance: no
gluetun  | |           └── Blocked IP networks:
gluetun  | |               ├── 127.0.0.1/8
gluetun  | |               ├── 10.0.0.0/8
gluetun  | |               ├── 172.16.0.0/12
gluetun  | |               ├── 192.168.0.0/16
gluetun  | |               ├── 169.254.0.0/16
gluetun  | |               ├── ::1/128
gluetun  | |               ├── fc00::/7
gluetun  | |               ├── fe80::/10
gluetun  | |               ├── ::ffff:127.0.0.1/104
gluetun  | |               ├── ::ffff:10.0.0.0/104
gluetun  | |               ├── ::ffff:169.254.0.0/112
gluetun  | |               ├── ::ffff:172.16.0.0/108
gluetun  | |               └── ::ffff:192.168.0.0/112
gluetun  | ├── Firewall settings:
gluetun  | |   └── Enabled: yes
gluetun  | ├── Log settings:
gluetun  | |   └── Log level: INFO
gluetun  | ├── Health settings:
gluetun  | |   ├── Server listening address: 127.0.0.1:9999
gluetun  | |   ├── Target address: cloudflare.com:443
gluetun  | |   ├── Duration to wait after success: 5s
gluetun  | |   ├── Read header timeout: 100ms
gluetun  | |   ├── Read timeout: 500ms
gluetun  | |   └── VPN wait durations:
gluetun  | |       ├── Initial duration: 6s
gluetun  | |       └── Additional duration: 5s
gluetun  | ├── Shadowsocks server settings:
gluetun  | |   └── Enabled: no
gluetun  | ├── HTTP proxy settings:
gluetun  | |   └── Enabled: no
gluetun  | ├── Control server settings:
gluetun  | |   ├── Listening address: :8000
gluetun  | |   └── Logging: yes
gluetun  | ├── OS Alpine settings:
gluetun  | |   ├── Process UID: 1000
gluetun  | |   └── Process GID: 1000
gluetun  | ├── Public IP settings:
gluetun  | |   ├── Fetching: every 12h0m0s
gluetun  | |   └── IP file path: /tmp/gluetun/ip
gluetun  | └── Version settings:
gluetun  |     └── Enabled: yes
gluetun  | 2023-11-10T16:05:27Z INFO [routing] default route found: interface eth0, gateway 172.30.128.1, assigned IP 172.30.128.2 and family v4
gluetun  | 2023-11-10T16:05:27Z INFO [routing] adding route for 0.0.0.0/0
gluetun  | 2023-11-10T16:05:27Z INFO [firewall] setting allowed subnets...
gluetun  | 2023-11-10T16:05:27Z INFO [routing] default route found: interface eth0, gateway 172.30.128.1, assigned IP 172.30.128.2 and family v4
gluetun  | 2023-11-10T16:05:27Z INFO [dns] using plaintext DNS at address 1.1.1.1
gluetun  | 2023-11-10T16:05:27Z INFO [http server] http server listening on [::]:8000
gluetun  | 2023-11-10T16:05:27Z INFO [healthcheck] listening on 127.0.0.1:9999
gluetun  | 2023-11-10T16:05:27Z INFO [firewall] allowing VPN connection...
gluetun  | 2023-11-10T16:05:27Z INFO [wireguard] Using available kernelspace implementation
gluetun  | 2023-11-10T16:05:27Z INFO [wireguard] Connecting to [REDACTED]
gluetun  | 2023-11-10T16:05:27Z INFO [wireguard] Wireguard setup is complete. Note Wireguard is a silent protocol and it may or may not work, without giving any error message. Typically i/o timeout errors indicate the Wireguard connection is not working.
gluetun  | 2023-11-10T16:05:27Z INFO [dns] downloading DNS over TLS cryptographic files
gluetun  | 2023-11-10T16:05:35Z INFO [healthcheck] program has been unhealthy for 6s: restarting VPN (see https://github.com/qdm12/gluetun-wiki/blob/main/faq/healthcheck.md)
gluetun  | 2023-11-10T16:05:35Z INFO [vpn] stopping
gluetun  | 2023-11-10T16:05:35Z ERROR [vpn] cannot get version information: Get "https://api.github.com/repos/qdm12/gluetun/commits": context canceled
gluetun  | 2023-11-10T16:05:35Z ERROR [ip getter] Get "https://ipinfo.io/": dial tcp: lookup ipinfo.io on 1.1.1.1:53: write udp 172.30.128.2:49829->1.1.1.1:53: write: operation not permitted - retrying in 5s
gluetun  | 2023-11-10T16:05:35Z INFO [vpn] starting
gluetun  | 2023-11-10T16:05:35Z INFO [firewall] allowing VPN connection...
gluetun  | 2023-11-10T16:05:35Z INFO [wireguard] Using available kernelspace implementation
gluetun  | 2023-11-10T16:05:35Z INFO [wireguard] Connecting to [REDACTED]
gluetun  | 2023-11-10T16:05:35Z INFO [wireguard] Wireguard setup is complete. Note Wireguard is a silent protocol and it may or may not work, without giving any error message. Typically i/o timeout errors indicate the Wireguard connection is not working.
^CGracefully stopping... (press Ctrl+C again to force)
[+] Running 1/1
 ⠿ Container gluetun  Stopped                                                                                                                                                                      0.8s
canceled
```

### Share your configuration

```yml
version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    container_name: gluetun
    restart: unless-stopped
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun:/dev/net/tun
    ports:
      - 8088:8888/tcp # HTTP proxy
      - 8388:8388/tcp # Shadowsocks
      - 8388:8388/udp # Shadowsocks
    volumes:
      - ./data:/gluetun
      - ./wg0.conf:/gluetun/wireguard/wg0.conf:ro
    environment:
      - VPN_SERVICE_PROVIDER=custom
      - VPN_TYPE=wireguard
```

**Comment by qdm12** [comment]
Can you run Gluetun with `LOG_LEVEL=debug` and comment back with the logs? Also make sure this is the same wireguard config for both docker images (and that the endpoint is the right/updated ip address) 😉 Thanks!

**Comment by diamondsw** [comment]
Thanks! I added the debug loglevel as requested. Confirmed that this is the exact same wg0.conf that works with the Linuxserver.io Wireguard container.

```
$ docker compose up
[+] Running 1/1
 ⠿ Container gluetun  Recreated                                                                                                                                                                    0.1s
Attaching to gluetun
gluetun  | ========================================
gluetun  | ========================================
gluetun  | =============== gluetun ================
gluetun  | ========================================
gluetun  | =========== Made with ❤️ by ============
gluetun  | ======= https://github.com/qdm12 =======
gluetun  | ========================================
gluetun  | ========================================
gluetun  | 
gluetun  | Running version latest built on 2023-11-08T10:11:52.918Z (commit 8318be3)
gluetun  | 
gluetun  | 🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
gluetun  | 🐛 Bug? https://github.com/qdm12/gluetun/issues/new
gluetun  | ✨ New feature? https://github.com/qdm12/gluetun/issues/new
gluetun  | ☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
gluetun  | 💻 Email? quentin.mcgaw@gmail.com
gluetun  | 💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
gluetun  | 2023-11-13T05:05:43Z INFO [routing] default route found: interface eth0, gateway 172.30.128.1, assigned IP 172.30.128.2 and family v4
gluetun  | 2023-11-13T05:05:43Z INFO [routing] local ethernet link found: eth0
gluetun  | 2023-11-13T05:05:43Z INFO [routing] local ipnet found: 172.30.128.0/24
gluetun  | 2023-11-13T05:05:43Z INFO [firewall] enabling...
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] iptables --policy INPUT DROP
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] iptables --policy OUTPUT DROP
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] iptables --policy FORWARD DROP
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] ip6tables-nft --policy INPUT DROP
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] ip6tables-nft --policy OUTPUT DROP
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] ip6tables-nft --policy FORWARD DROP
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] iptables --append INPUT -i lo -j ACCEPT
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] ip6tables-nft --append INPUT -i lo -j ACCEPT
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] iptables --append OUTPUT -o lo -j ACCEPT
gluetun  | 2023-11-13T05:05:43Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o lo -j ACCEPT
gluetun  | 2023-11-13T05:05:44Z DEBUG [firewall] iptables --append OUTPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
gluetun  | 2023-11-13T05:05:44Z DEBUG [firewall] ip6tables-nft --append OUTPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
gluetun  | 2023-11-13T05:05:44Z DEBUG [firewall] iptables --append INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
gluetun  | 2023-11-13T05:05:44Z DEBUG [firewall] ip6tables-nft --append INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
gluetun  | 2023-11-13T05:05:44Z DEBUG [firewall] iptables --append OUTPUT -o eth0 -s 172.30.128.2 -d 172.30.128.0/24 -j ACCEPT
gluetun  | 2023-11-13T05:05:44Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o eth0 -d ff02::1:ff/104 -j ACCEPT
gluetun  | 2023-11-13T05:05:44Z DEBUG [firewall] iptables --append INPUT -i eth0 -d 172.30.128.0/24 -j ACCEPT
gluetun  | 2023-11-13T05:05:44Z INFO [firewall] enabled successfully
gluetun  | 2023-11-13T05:05:45Z INFO [storage] merging by most recent 17689 hardcoded servers and 17689 servers read from /gluetun/servers.json
gluetun  | 2023-11-13T05:05:45Z DEBUG [netlink] IPv6 is not supported after searching 0 routes
gluetun  | 2023-11-13T05:05:45Z INFO Alpine version: 3.18.4
gluetun  | 2023-11-13T05:05:45Z INFO OpenVPN 2.5 version: 2.5.8
gluetun  | 2023-11-13T05:05:45Z INFO OpenVPN 2.6 version: 2.6.5
gluetun  | 2023-11-13T05:05:45Z INFO Unbound version: 1.17.1
gluetun  | 2023-11-13T05:05:45Z INFO IPtables version: v1.8.9
gluetun  | 2023-11-13T05:05:45Z INFO Settings summary:
gluetun  | ├── VPN settings:
gluetun  | |   ├── VPN provider settings:
gluetun  | |   |   ├── Name: custom
gluetun  | |   |   └── Server selection settings:
gluetun  | |   |       ├── VPN type: wireguard
gluetun  | |   |       └── Wireguard selection settings:
gluetun  | |   |           ├── Endpoint IP address: [REDACTED]
gluetun  | |   |           ├── Endpoint port: [REDACTED]
gluetun  | |   |           └── Server public key: [REDACTED]
gluetun  | |   └── Wireguard settings:
gluetun  | |       ├── Private key: [REDACTED]
gluetun  | |       ├── Interface addresses:
gluetun  | |       |   └── 10.13.13.2/32
gluetun  | |       ├── Allowed IPs:
gluetun  | |       |   ├── 0.0.0.0/0
gluetun  | |       |   └── ::/0
gluetun  | |       └── Network interface: tun0
gluetun  | |           └── MTU: 1400
gluetun  | ├── DNS settings:
gluetun  | |   ├── Keep existing nameserver(s): no
gluetun  | |   ├── DNS server address to use: 127.0.0.1
gluetun  | |   └── DNS over TLS settings:
gluetun  | |       ├── Enabled: yes
gluetun  | |       ├── Update period: every 24h0m0s
gluetun  | |       ├── Unbound settings:
gluetun  | |       |   ├── Authoritative servers:
gluetun  | |       |   |   └── cloudflare
gluetun  | |       |   ├── Caching: yes
gluetun  | |       |   ├── IPv6: no
gluetun  | |       |   ├── Verbosity level: 1
gluetun  | |       |   ├── Verbosity details level: 0
gluetun  | |       |   ├── Validation log level: 0
gluetun  | |       |   ├── System user: root
gluetun  | |       |   └── Allowed networks:
gluetun  | |       |       ├── 0.0.0.0/0
gluetun  | |       |       └── ::/0
gluetun  | |       └── DNS filtering settings:
gluetun  | |           ├── Block malicious: yes
gluetun  | |           ├── Block ads: no
gluetun  | |           ├── Block surveillance: no
gluetun  | |           └── Blocked IP networks:
gluetun  | |               ├── 127.0.0.1/8
gluetun  | |               ├── 10.0.0.0/8
gluetun  | |               ├── 172.16.0.0/12
gluetun  | |               ├── 192.168.0.0/16
gluetun  | |               ├── 169.254.0.0/16
gluetun  | |               ├── ::1/128
gluetun  | |               ├── fc00::/7
gluetun  | |               ├── fe80::/10
gluetun  | |               ├── ::ffff:127.0.0.1/104
gluetun  | |               ├── ::ffff:10.0.0.0/104
gluetun  | |               ├── ::ffff:169.254.0.0/112
gluetun  | |               ├── ::ffff:172.16.0.0/108
gluetun  | |               └── ::ffff:192.168.0.0/112
gluetun  | ├── Firewall settings:
gluetun  | |   └── Enabled: yes
gluetun  | ├── Log settings:
gluetun  | |   └── Log level: DEBUG
gluetun  | ├── Health settings:
gluetun  | |   ├── Server listening address: 127.0.0.1:9999
gluetun  | |   ├── Target address: cloudflare.com:443
gluetun  | |   ├── Duration to wait after success: 5s
gluetun  | |   ├── Read header timeout: 100ms
gluetun  | |   ├── Read timeout: 500ms
gluetun  | |   └── VPN wait durations:
gluetun  | |       ├── Initial duration: 6s
gluetun  | |       └── Additional duration: 5s
gluetun  | ├── Shadowsocks server settings:
gluetun  | |   └── Enabled: no
gluetun  | ├── HTTP proxy settings:
gluetun  | |   └── Enabled: no
gluetun  | ├── Control server settings:
gluetun  | |   ├── Listening address: :8000
gluetun  | |   └── Logging: yes
gluetun  | ├── OS Alpine settings:
gluetun  | |   ├── Process UID: 1000
gluetun  | |   └── Process GID: 1000
gluetun  | ├── Public IP settings:
gluetun  | |   ├── Fetching: every 12h0m0s
gluetun  | |   └── IP file path: /tmp/gluetun/ip
gluetun  | └── Version settings:
gluetun  |     └── Enabled: yes
gluetun  | 2023-11-13T05:05:45Z INFO [routing] default route found: interface eth0, gateway 172.30.128.1, assigned IP 172.30.128.2 and family v4
gluetun  | 2023-11-13T05:05:45Z DEBUG [routing] ip rule add from 172.30.128.2/32 lookup 200 pref 100
gluetun  | 2023-11-13T05:05:45Z INFO [routing] adding route for 0.0.0.0/0
gluetun  | 2023-11-13T05:05:45Z DEBUG [routing] ip route replace 0.0.0.0/0 via 172.30.128.1 dev eth0 table 200
gluetun  | 2023-11-13T05:05:45Z INFO [firewall] setting allowed subnets...
gluetun  | 2023-11-13T05:05:45Z INFO [routing] default route found: interface eth0, gateway 172.30.128.1, assigned IP 172.30.128.2 and family v4
gluetun  | 2023-11-13T05:05:45Z DEBUG [routing] ip rule add to 172.30.128.0/24 lookup 254 pref 98
gluetun  | 2023-11-13T05:05:45Z INFO [dns] using plaintext DNS at address 1.1.1.1
gluetun  | 2023-11-13T05:05:45Z INFO [http server] http server listening on [::]:8000
gluetun  | 2023-11-13T05:05:45Z DEBUG [wireguard] Wireguard server public key: [REDACTED]
gluetun  | 2023-11-13T05:05:45Z DEBUG [wireguard] Wireguard client private key: [REDACTED]
gluetun  | 2023-11-13T05:05:45Z DEBUG [wireguard] Wireguard pre-shared key: [not set]
gluetun  | 2023-11-13T05:05:45Z INFO [healthcheck] listening on 127.0.0.1:9999
gluetun  | 2023-11-13T05:05:45Z INFO [firewall] allowing VPN connection...
gluetun  | 2023-11-13T05:05:45Z DEBUG [firewall] iptables --append OUTPUT -d [REDACTED] -o eth0 -p udp -m udp --dport [REDACTED] -j ACCEPT
gluetun  | 2023-11-13T05:05:45Z DEBUG [firewall] iptables --append OUTPUT -o tun0 -j ACCEPT
gluetun  | 2023-11-13T05:05:45Z DEBUG [firewall] ip6tables-nft --append OUTPUT -o tun0 -j ACCEPT
gluetun  | 2023-11-13T05:05:45Z INFO [wireguard] Using available kernelspace implementation
gluetun  | 2023-11-13T05:05:45Z INFO [wireguard] Connecting to [REDACTED]:[REDACTED]
gluetun  | 2023-11-13T05:05:45Z INFO [wireguard] Wireguard setup is complete. Note Wireguard is a silent protocol and it may or may not work, without giving any error message. Typically i/o timeout errors indicate the Wireguard connection is not working.
gluetun  | 2023-11-13T05:05:46Z INFO [dns] downloading DNS over TLS cryptographic files
gluetun  | 2023-11-13T05:05:53Z INFO [healthcheck] program has been unhealthy for 6s: restarting VPN (see https://github.com/qdm12/gluetun-wiki/blob/main/faq/healthcheck.md)
gluetun  | 2023-11-13T05:05:53Z INFO [vpn] stopping
gluetun  | 2023-11-13T05:05:53Z DEBUG [wireguard] closing controller client...
gluetun  | 2023-11-13T05:05:53Z DEBUG [wireguard] removing IPv4 rule...
gluetun  | 2023-11-13T05:05:53Z DEBUG [wireguard] shutting down link...
gluetun  | 2023-11-13T05:05:53Z ERROR [vpn] cannot get version information: Get "https://api.github.com/repos/qdm12/gluetun/commits": context canceled
gluetun  | 2023-11-13T05:05:53Z DEBUG [wireguard] deleting link...
gluetun  | 2023-11-13T05:05:53Z ERROR [ip getter] Get "https://ipinfo.io/": dial tcp: lookup ipinfo.io on 1.1.1.1:53: write udp 172.30.128.2:43104->1.1.1.1:53: write: operation not permitted - retrying in 5s
gluetun  | 2023-11-13T05:05:54Z INFO [vpn] starting
gluetun  | 2023-11-13T05:05:54Z DEBUG [wireguard] Wireguard server public key: [REDACTED]
gluetun  | 2023-11-13T05:05:54Z DEBUG [wireguard] Wireguard client private key: [REDACTED]
gluetun  | 2023-11-13T05:05:54Z DEBUG [wireguard] Wireguard pre-shared key: [not set]
gluetun  | 2023-11-13T05:05:54Z INFO [firewall] allowing VPN connection...
gluetun  | 2023-11-13T05:05:54Z INFO [wireguard] Using available kernelspace implementation
gluetun  | 2023-11-13T05:05:54Z INFO [wireguard] Connecting to [REDACTED]:[REDACTED]
gluetun  | 2023-11-13T05:05:54Z INFO [wireguard] Wireguard setup is complete. Note Wireguard is a silent protocol and it may or may not work, without giving any error message. Typically i/o timeout errors indicate the Wireguard connection is not working.
^CGracefully stopping... (press Ctrl+C again to force)
[+] Running 1/1
 ⠿ Container gluetun  Stopped                                                                                                                                                                      0.5s
canceled
```

**Comment by diamondsw** [comment]
Is there anything else I need to provide, or anything that looks amiss? I'm willing to privately provide access to the VPN endpoint if it's needed for troubeshooting.

**Comment by qdm12** [comment]
It looks fine 🤔 

> I'm willing to privately provide access to the VPN endpoint if it's needed for troubleshooting.

Sure, send me an email to quentin.mcgaw@gmail.com with:

- the same wireguard config file
- the link to this issue

And I'll try it over here, then you can rotate the Wireguard keys 😉

**Comment by qdm12** [comment]
> Identical wg0.conf connects without issue using the Linuxserver.io image

Actually, it doesn't error out since it has no health check and Wireguard won't return an error if no traffic can go through, but can you tunnel through it and get the right tunnel ip address? As in, does it **actually** work??

**Comment by diamondsw** [comment]
> > Identical wg0.conf connects without issue using the Linuxserver.io image
> 
> Actually, it doesn't error out since it has no health check and Wireguard won't return an error if no traffic can go through, but can you tunnel through it and get the right tunnel ip address? As in, does it **actually** work??

As I recall I could, and just went and tested it to be sure:
```
docker wireguard $ docker compose up -d
[+] Running 8/8
 ⠿ wireguard Pulled                                                                                                                                                                                4.3s
   ⠿ c0fc586c2f6e Already exists                                                                                                                                                                   0.0s
   ⠿ 07a0e16f7be1 Already exists                                                                                                                                                                   0.0s
   ⠿ 6b85830a6b99 Already exists                                                                                                                                                                   0.0s
   ⠿ 3088b9d135b1 Already exists                                                                                                                                                                   0.0s
   ⠿ 8dfc225d7840 Already exists                                                                                                                                                                   0.0s
   ⠿ bf98b3ec32c5 Pull complete                                                                                                                                                                    3.0s
   ⠿ aaf0093d7a89 Pull complete                                                                                                                                                                    3.2s
[+] Running 2/2
 ⠿ Network wireguard_default  Created                                                                                                                                                              0.4s
 ⠿ Container wireguard        Started                                                                                                                                                              1.4s
docker wireguard $ docker exec -it wireguard wget -qO- http://ipecho.net/plain | xargs echo
[REDACTED]
docker wireguard $ 
```

I know the [REDACTED] is a bit silly, but yes - it's able to pull down an IP check from the far side of the tunnel.

**Comment by qdm12** [comment]
Found the problem (thanks to your config sent by email 😉):

```
DEBUG [wireguard] Wireguard pre-shared key: [not set]
``` 

In your config, there is a preshared key so you need to set it with `WIREGUARD_PRESHARED_KEY` 😉 It succeeded the first time I launched it, so it should work fine on your end with this config fix. Feel free to rotate your Wireguard keys too now. I'll close this, if you encounter another problem please open another issue (I don't monitor closed issues), thanks!

Happy tunneling!

**Comment by qdm12** [comment]
Oh oh also related and just-freshly-fixed: #1997 with commit 657b4b787f247d6f6c5180ca4766b2c97dba6ecf
It was trying to read the presharedkey from the interface section instead of the peer section in the wireguard config file (env variables worked fine though).
